experiment/commenter/main.go

```diff
@@ -0,0 +1,204 @@
+	mat := re.FindStringSubmatch(url)
+	if mat == nil {
+		return "", "", 0, fmt.Errorf("failed to parse: %s", url)
+	}
+	n, err := strconv.Atoi(mat[4])
+	if err != nil {
+		return "", "", 0, err
+	}
+	return mat[1], mat[2], n, nil
+}
+
+func makeQuery(query string, includeClosed bool, minUpdated time.Duration) string {
+	parts := []string{query}
+	if !includeClosed {
+		parts = append(parts, "is:open")
+	}
+	if minUpdated != 0 {
+		latest := time.Now().Add(-minUpdated)
+		parts = append(parts, "updated:<="+latest.Format(time.RFC3339))
+	}
+	return strings.Join(parts, " ")
+}
+
+type client interface {
+	CreateComment(owner, repo string, number int, comment string) error
+	FindIssues(query, sort string, asc bool) ([]github.Issue, error)
+}
+
+func main() {
+	log.SetFlags(log.LstdFlags | log.Lshortfile)
+	o := flagOptions()
+
+	if o.query == "" {
+		log.Fatal("empty --query")
+	}
+	if o.user == "" {
+		log.Fatal("empty --user")
+	}
+	if o.token == "" {
+		log.Fatal("empty --token")
+	}
+	b, err := ioutil.ReadFile(o.token)
+	if err != nil {
+		log.Fatalf("cannot read --token: %v", err)
+	}
+
+	var c client
+	tok := strings.TrimSpace(string(b))
+	if o.wet {
+		c = github.NewClient(o.user, tok)
+	} else {
+		c = github.NewDryRunClient(o.user, tok)
+	}
+
+	query := makeQuery(o.query, o.includeClosed, o.updated)
+	sort := ""
+	asc := false
+	if o.updated > 0 {
+		sort = "updated"
+		asc = true
+	}
+	commenter := makeCommenter(o.comment, o.useTemplate)
+	if err := run(c, query, sort, asc, commenter, o.ceiling); err != nil {
+		log.Fatalf("Failed run: %v", err)
+	}
+}
+
+func makeCommenter(comment string, useTemplate bool) func(meta) (string, error) {
+	if comment == "" || !useTemplate {
+		return func(_ meta) (string, error) {
+			return comment, nil
+		}
+	}
+	t := template.Must(template.New("comment").Parse(comment))
+	return func(m meta) (string, error) {
+		out := bytes.Buffer{}
+		err := t.Execute(&out, m)
+		return string(out.Bytes()), err
+	}
+}
+
+func run(c client, query, sort string, asc bool, commenter func(meta) (string, error), ceiling int) error {
+	log.Printf("Searching: %s", query)
+	issues, err := c.FindIssues(query, sort, asc)
+	if err != nil {
+		return fmt.Errorf("search failed: %v", err)
+	}
+	problems := []string{}
+	log.Printf("Found %d matches", len(issues))
+	for n, i := range issues {
+		if ceiling > 0 && n == ceiling {
+			log.Printf("Stopping at --ceiling=%d of %d results", n, len(issues))
+			break
+		}
+		log.Printf("Matched %s: %s", i.HTMLURL, i.Title)
+		org, repo, number, err := parseHTMLURL(i.HTMLURL)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to parse %s: %v", i.HTMLURL, err)
+			log.Print(msg)
+			problems = append(problems, msg)
+		}
+		if commenter == nil {
+			continue
+		}
+		comment, err := commenter(meta{Number: number, Org: org, Repo: repo, Issue: i})
+		if err != nil {
+			msg := fmt.Sprintf("Failed to create comment for %s/%s#%d: %v", org, repo, number, err)
+			log.Print(msg)
```

Comment:
It's a simple tool so I guess this is fine. We used to duplicate a lot of info in the controller manager logs (and still do to some extend) that made it really annoying when I needed to debug something.